.github/workflows/prettier.yml

```diff
@@ -0,0 +1,38 @@
+name: check prettier formatting
+
+on:
+  workflow_dispatch:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+    branches:
+      - "main"
+
+concurrency:
+  # Cancel previous jobs
+  group: prettier-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  prettier:
+    runs-on: ubuntu-latest
+    steps:
+      - name: check out repository
+        uses: actions/checkout@v4
+      - name: set up node
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: front_end/.nvmrc
+          cache: npm
+          cache-dependency-path: front_end/package-lock.json
+      - name: install deps
+        run: npm ci
+        working-directory: front_end
+      - name: prettier version
+        run: npx prettier -v
+        working-directory: front_end
+      - name: check formatting
+        run: npm run format:check
+        working-directory: front_end
```

Comment:
Could we also add a step to run linters and typescript checks (`npm run lint`)?

Reply:
can do - I'd rather follow up in a separate PR though if you don't mind - for the reason of consideration (1) above, and also to stop the PR getting too big (I took a look at what it'd take to run linter on everything and there were a good number of changes to solve the errors when I last checked)

Reply:
Yeah, makes sense to do in a separate PR 👍